junit5-extension/src/test/java/io/kroxylicious/testing/kafka/junit5ext/ParameterExtensionTest.java

```diff
@@ -141,11 +142,25 @@ public void kraftBasedClusterParameter(@BrokerCluster @KRaftCluster KafkaCluster
     }
 
     @Test
-    public void saslPlainAuthenticatingClusterParameter(
-                                                        @BrokerCluster @SaslPlainAuth({
-                                                                @SaslPlainAuth.UserPassword(user = "alice", password = "foo"),
-                                                                @SaslPlainAuth.UserPassword(user = "bob", password = "bar")
-                                                        }) KafkaCluster cluster)
+    public void saslPlainAuthenticatingClusterParameterOneUser(
+                                                               @BrokerCluster @SaslPlainAuth @User(user = "alice", password = "foo") KafkaCluster cluster)
+            throws ExecutionException, InterruptedException {
+        var dc = describeCluster(cluster.getKafkaClientConfiguration("alice", "foo"));
+        assertEquals(1, dc.nodes().get().size());
+        assertEquals(cluster.getClusterId(), dc.clusterId().get());
+
+        var ee = assertThrows(ExecutionException.class, () -> describeCluster(cluster.getKafkaClientConfiguration("alice", "FOO")),
+                "Expect bad password to throw");
+        assertInstanceOf(SaslAuthenticationException.class, ee.getCause());
+
+        ee = assertThrows(ExecutionException.class, () -> describeCluster(cluster.getKafkaClientConfiguration("eve", "quux")),
+                "Expect unknown user to throw");
+        assertInstanceOf(SaslAuthenticationException.class, ee.getCause());
+    }
+
+    @Test
+    public void saslPlainAuthenticatingClusterParameterTwoUsers(
+                                                                @BrokerCluster @SaslPlainAuth @User(user = "alice", password = "foo") @User(user = "bob", password = "bar") KafkaCluster cluster)
```

Comment:
I found the old version easier to think about. 

```java
@SaslPlainAuth({
                                                                 @SaslPlainAuth.UserPassword(user = "alice", password = "foo"),
                                                                 @SaslPlainAuth.UserPassword(user = "bob", password = "bar")
                                                         }
```

Clearly says to me that these users are authenticated via Sasl Plain. Where as the new version you can assume the same but it's less explicit.